File: C/rand32.c
#include "rand/rand32.h"

rand32_t rand32_init(int seed)
{
    rand32_t ret;
    ret.seed = abs(seed) % 9999999 + 1;
    ret.curr = ret.seed;
    ret.iter = 0;
    return ret;
}

int rand32_range(rand32_t* rand, int min, int max)
{
    rand->curr = (rand->curr * 125 ) % 2796203;
	rand->iter = rand->iter + 1;
    return rand->curr % (max - min + 1) + min;
}

int rand32_rand(rand32_t* rand, int max)
{
    return rand32_range(rand, 0, max);
}

File: C#/Source/Rand/Rand32.cs
using System;

namespace Rand
{
    public class Rand32
    {
        public int Seed;
        public int Current;
        public int Iterator;

        public Rand32(int Seed)
        {
            this.Seed = (Math.Abs(Seed)) % 9999999 + 1;
            this.Current = this.Seed;
            this.Iterator = 0;
        }

        public int Range(int Minimum, int Maximum)
        {
            this.Current = ( this.Current * 125 ) % 2796203;
            this.Iterator = this.Iterator + 1;
            return this.Current % (Maximum - Minimum + 1) + Minimum;
        }

        public int Rand(int Maximum)
        {
            return this.Range(0, Maximum);
        }
    }
}
